pkg/metrics/prometheus_exporter.go

```diff
@@ -0,0 +1,72 @@
+package metrics
+
+import (
+	"fmt"
+	"net/http"
+	"sync"
+
+	"contrib.go.opencensus.io/exporter/prometheus"
+	"go.opencensus.io/stats/view"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+var (
+	curPromSrv    *http.Server
+	curPromSrvMux sync.Mutex
+)
+
+var log = logf.Log.WithName("metrics")
+
+const namespace = "gatekeeper"
+
+func newPrometheusExporter() (view.Exporter, error) {
+	e, err := prometheus.NewExporter(prometheus.Options{Namespace: namespace})
+	if err != nil {
+		log.Error(err, "Failed to create the Prometheus exporter.")
+		return nil, err
+	}
+	errCh := make(chan error)
+	log.Info("Starting server for OpenCensus Prometheus exporter")
+	// Start the server for Prometheus scraping
+	go func() {
+		srv := startNewPromSrv(e, *prometheusPort)
+		errCh <- srv.ListenAndServe()
+	}()
+	select {
+	case err := <-errCh:
+		if err != nil {
+			return e, err
+		}
+	}
+	return e, nil
```

Comment:
There is no need to return the exporter if this function is blocking until the exporter errors out or finishes anyway

Reply:
changed to return nil, nil